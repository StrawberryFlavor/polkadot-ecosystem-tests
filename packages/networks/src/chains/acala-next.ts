import { acala, karura } from './acala.js'

import path from 'path'

export const acalaNext = acala.extend(() => ({
  name: 'acalaNext',
  wasmOverride: path.resolve(__dirname, '../wasm/acala_runtime.wasm'),
}))

export const karuraNext = karura.extend(() => ({
  name: 'karuraNext',
  wasmOverride: path.resolve(__dirname, '../wasm/karura_runtime.wasm'),
}))
