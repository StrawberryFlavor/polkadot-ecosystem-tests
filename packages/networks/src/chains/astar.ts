import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccount.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccount.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'https://rpc.astar.network',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'https://rpc.shiden.astar.network',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
