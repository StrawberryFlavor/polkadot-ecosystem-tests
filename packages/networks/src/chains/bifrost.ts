import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  bifrostPolkadot: {
    relayToken: 'DOT',
  },
  bifrost: {
    relayToken: 'KSM',
  },
}

const getInitStorages = (config: typeof custom.bifrostPolkadot | typeof custom.bifrost) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 1, data: { free: '1000000000000000' } }]],
  },
  Tokens: {
    account: [[[defaultAccount.alice.address, config.relayToken], { balance: 10 * 1e12 }]],
  },
})

export const bifrostPolkadot = defineChain({
  name: 'bifrostPolkadot',
  paraId: 2030,
  endpoint: 'wss://hk.p.bifrost-rpc.liebi.com/ws',
  custom: custom.bifrostPolkadot,
  initStorages: getInitStorages(custom.bifrostPolkadot),
})

export const bifrost = defineChain({
  name: 'bifrost',
  paraId: 2001,
  endpoint: 'wss://bifrost-rpc.liebi.com/ws',
  custom: custom.bifrost,
  initStorages: getInitStorages(custom.bifrost),
})
