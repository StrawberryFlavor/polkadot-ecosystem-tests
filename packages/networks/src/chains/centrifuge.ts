import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  centrifuge: {
    acalaUsd: 'AUSD',
  },
  altair: {
    acalaUsd: 'KUSD',
  },
}

const getInitStorages = (config: typeof custom.centrifuge | typeof custom.altair) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 1, data: { free: '1000000000000000' } }]],
  },
  OrmlTokens: {
    accounts: [[[defaultAccount.alice.address, config.acalaUsd], { balance: 10 * 1e12 }]],
  },
})

export const centrifuge = defineChain({
  name: 'centrifuge',
  paraId: 2031,
  endpoint: 'wss://fullnode.centrifuge.io',
  custom: custom.centrifuge,
  initStorages: getInitStorages(custom.centrifuge),
})

export const altair = defineChain({
  name: 'altair',
  paraId: 2088,
  endpoint: 'wss://fullnode.altair.centrifuge.io',
  custom: custom.altair,
  initStorages: getInitStorages(custom.altair),
})
