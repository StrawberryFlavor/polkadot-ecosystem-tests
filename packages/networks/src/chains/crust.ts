import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  crustPolkadot: {},
  crust: {},
}

const getInitStorages = () => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 1, data: { free: '1000000000000000' } }]],
  },
})

export const crustPolkadot = defineChain({
  name: 'crustPolkadot',
  paraId: 2008,
  endpoint: 'wss://crust-parachain.crustapps.net',
  custom: custom.crustPolkadot,
  initStorages: getInitStorages(),
})

export const crust = defineChain({
  name: 'crust',
  paraId: 2012,
  endpoint: 'wss://rpc-shadow.crust.network/',
  custom: custom.crust,
  initStorages: getInitStorages(),
})
