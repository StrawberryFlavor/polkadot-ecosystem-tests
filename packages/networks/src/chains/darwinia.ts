import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  darwinia: {},
  crab: {},
}

const getInitStorages = () => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 1, data: { free: '1000000000000000' } }]],
  },
})

export const darwinia = defineChain({
  name: 'darwinia',
  paraId: 2046,
  endpoint: 'wss://darwinia-rpc.dwellir.com',
  custom: custom.darwinia,
  initStorages: getInitStorages(),
})

export const crab = defineChain({
  name: 'crab',
  paraId: 2105,
  endpoint: 'wss://crab-rpc.darwinia.network',
  custom: custom.crab,
  initStorages: getInitStorages(),
})
