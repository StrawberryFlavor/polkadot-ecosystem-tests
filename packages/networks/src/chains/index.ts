export * from './acala-next.js'
export { acala, karura } from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
