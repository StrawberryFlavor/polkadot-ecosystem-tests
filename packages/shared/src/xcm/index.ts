export * from './runXcmPalletDown.js'
export * from './runXcmPalletHorizontal.js'
export * from './runXtokensUp.js'
export * from './runXtokenstHorizontal.js'
