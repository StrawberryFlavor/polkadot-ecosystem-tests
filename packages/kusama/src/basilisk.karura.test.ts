import { afterAll, describe } from 'vitest'
import { connectParachains } from '@acala-network/chopsticks'

import { basilisk, karura } from '@e2e-test/networks/chains'
import { createNetwork } from '@e2e-test/networks'
import { query, tx } from '@e2e-test/shared/api'
import { runXcmPalletHorizontal } from '@e2e-test/shared/xcm'

// karura <=> basilisk
describe(`'karura' <-> 'basilisk' xcm transfer 'DAI'`, async () => {
  const [karuraClient, basiliskClient] = await Promise.all([createNetwork(karura), createNetwork(basilisk)])

  await connectParachains([karuraClient.chain, basiliskClient.chain])

  const karuraDai = karuraClient.config.custom!.dai.Erc20
  const basiliskDai = basiliskClient.config.custom!.dai

  afterAll(async () => {
    await karuraClient.teardown()
    await basiliskClient.teardown()
  })

  runXcmPalletHorizontal(`'karura' -> 'basilisk' xcm transfer 'DAI'`, async () => {
    await karuraClient.dev.setStorage({
      Evm: {
        accountStorages: [
          [
            [
              karuraDai,
              '0x2aef47e62c966f0695d5af370ddc1bc7c56902063eee60853e2872fc0ff4f88c', // balanceOf(Alice)
            ],
            '0x0000000000000000000000000000000000000000000000056bc75e2d63100000', // 1e20
          ],
        ],
      },
    })

    return {
      fromChain: karuraClient,
      toChain: basiliskClient,
      fromBalance: query.evm(karuraDai, '0x2aef47e62c966f0695d5af370ddc1bc7c56902063eee60853e2872fc0ff4f88c'),
      toBalance: query.tokens(basiliskDai),
      tx: tx.xtokens.transfer(karura.dai, 10n ** 18n, tx.xtokens.parachainV3(basiliskClient.config.paraId!)),
    }
  })

  // TODO: restore this once Basilisk fixed the asset mapping issue
  //   runXcmPalletHorizontal(`'basilisk' -> 'karura' xcm transfer 'DAI'`, async () => {
  //     await basiliskClient.dev.setStorage({
  //       Tokens: {
  //         accounts: [[[defaultAccount.alice.address, basiliskDai], { free: 10n * 10n ** 18n }]],
  //       },
  //     })

  //     return {
  //       fromChain: basiliskClient,
  //       toChain: karuraClient,
  //       fromBalance: query.tokens(basiliskDai),
  //       toBalance: query.evm(karuraDai, '0x2aef47e62c966f0695d5af370ddc1bc7c56902063eee60853e2872fc0ff4f88c'),
  //       tx: tx.xtokens.transfer(basilisk.dai, 10n ** 18n, tx.xtokens.parachainV3(karuraClient.config.paraId!)),
  //     }
  //   })
})
