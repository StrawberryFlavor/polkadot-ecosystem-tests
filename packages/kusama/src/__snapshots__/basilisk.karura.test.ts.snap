// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`'karura' <-> 'basilisk' xcm transfer 'DAI' > 'karura' -> 'basilisk' xcm transfer 'DAI' > balance on from chain 1`] = `100000000000000000000`;

exports[`'karura' <-> 'basilisk' xcm transfer 'DAI' > 'karura' -> 'basilisk' xcm transfer 'DAI' > balance on to chain 1`] = `
{
  "free": 100000000000000000000,
  "frozen": 0,
  "reserved": 0,
}
`;

exports[`'karura' <-> 'basilisk' xcm transfer 'DAI' > 'karura' -> 'basilisk' xcm transfer 'DAI' > from chain hrmp messages 1`] = `[]`;

exports[`'karura' <-> 'basilisk' xcm transfer 'DAI' > 'karura' -> 'basilisk' xcm transfer 'DAI' > to chain xcm events 1`] = `[]`;

exports[`'karura' <-> 'basilisk' xcm transfer 'DAI' > 'karura' -> 'basilisk' xcm transfer 'DAI' > tx events 1`] = `[]`;
